mrjob/job.py

```diff
@@ -67,6 +63,20 @@
     'org.apache.hadoop.mapred.lib.KeyFieldBasedPartitioner'
 
 
+def _im_func(f):
+    """Wrapper to get at the underlying function belonging to a method.
+
+    Python 2 is slightly different because classes have "unbound methods"
+    which wrap the underlying function, whereas on Python 3 they're just
+    functions. (Methods work the same way on both versions.)
+    """
+    # "im_func" is the old Python 2 name for __func__
+    if hasattr(f, '__func__'):
+        return f.__func__
+    else:
+        return f
```

Comment:
Again, not going to add `six` as a dependency, but this is one place where I think mine is a little bit better; it's simple, and it can handle methods, unbound methods, and regular functions.

This function is really only used in one place (two, if you count tests), which is why I left it in `job.py`.